src/context.rs

```diff
@@ -89,8 +101,16 @@ impl TryFrom<Args> for KrunContext {
 
         set_smbios_oem_strings(id, &args.oem_strings)?;
 
-        if args.nested && unsafe { krun_set_nested_virt(id, args.nested) } < 0 {
-            return Err(anyhow!("unable to set krun nested virtualization"));
+        if args.nested {
+            match unsafe { krun_check_nested_virt() } {
+                1 => {
+                    if unsafe { krun_set_nested_virt(id, args.nested) } < 0 {
+                        return Err(anyhow!("unable to set krun nested virtualization"));
```

Comment:
```suggestion
                        return Err(anyhow!("krun nested virtualization reported as supported, yet unable to enable"));
```

Something showing that it's technically supported yet for some reason couldn't actually be enabled.

Reply:
Fixed!